akka-operator/templates/040-deployment.yaml

```diff
@@ -79,22 +79,35 @@ spec:
       labels:
         {{- include "akka-operator.selectorLabels" . | nindent 8 }}
     spec:
-      {{- with .Values.imagePullSecrets }}
-      imagePullSecrets:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
       serviceAccountName: {{ include "akka-operator.serviceAccountName" . }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- if ne .Values.lightbendSubscription.username "false" }}
+      imagePullSecrets:
+        - name: commercial-credentials
+      {{- else }}
+        {{- with .Values.imagePullSecrets }}
+        imagePullSecrets:
+          {{- toYaml . | nindent 8 }}
+        {{- end }}
+      {{- end }}
       containers:
         - name: {{ .Chart.Name }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
-          {{- if and (eq (toString .Values.provider.name) "gcp") (eq (toString .Values.gcp.deployer) "false") }}
+          # There's several possible registries to pull the image from:
+          #  1. lightbend commercial (picked when the user provides subscription credentials)
+          #  2. GCP's (multiple alternatives)
+          #  3. AWS marketplace
+          {{- if ne .Values.lightbendSubscription.username "false" }}
+          image: "{{ .Values.lightbendSubscription.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
```

Comment:
Note how the image is pulled from the `repository` (a field on `lightbendSubscription`).